### PR TITLE
feat(stoppers): implement ScoreThresholdStopper

### LINKED ISSUE #193: [Feature] ScoreThresholdStopper implementation
## Parent Issue
Part of #51 - Pluggable stop conditions

## User Story
**As a** gepa-adk user, **I want** to stop evolution when a target score is reached, **so that** I don't waste compute once "good enough" performance is achieved.

## Description
Implement a stopper that terminates evolution when the best score meets or exceeds a threshold. This enables "early success" termination.

### Files to Create/Modify

**New: `src/gepa_adk/adapters/stoppers/threshold.py`**
```python
from gepa_adk.domain.stopper import StopperState
from gepa_adk.ports.stopper import StopperProtocol

class ScoreThresholdStopper:
    """Stop evolution when best score reaches threshold.
    
    Args:
        threshold: Target score to achieve (evolution stops when best_score >= threshold)
        
    Example:
        >>> stopper = ScoreThresholdStopper(0.95)  # Stop at 95% accuracy
        >>> config = EvolutionConfig(stop_callbacks=[stopper])
    """
    
    def __init__(self, threshold: float) -> None:
        self.threshold = threshold
    
    def __call__(self, state: StopperState) -> bool:
        return state.best_score >= self.threshold
```

**Update: `src/gepa_adk/adapters/stoppers/__init__.py`**
```python
from gepa_adk.adapters.stoppers.timeout import TimeoutStopper
from gepa_adk.adapters.stoppers.threshold import ScoreThresholdStopper

__all__ = ["TimeoutStopper", "ScoreThresholdStopper"]
```

## Acceptance Criteria

```gherkin
Scenario: Stop when threshold reached
  Given ScoreThresholdStopper configured with threshold=0.9
  And StopperState with best_score=0.92
  When stopper is called
  Then it returns True

Scenario: Stop when threshold exactly met
  Given ScoreThresholdStopper configured with threshold=0.9
  And StopperState with best_score=0.9
  When stopper is called
  Then it returns True

Scenario: Continue below threshold
  Given ScoreThresholdStopper configured with threshold=0.9
  And StopperState with best_score=0.85
  When stopper is called
  Then it returns False

Scenario: Handle negative scores
  Given ScoreThresholdStopper configured with threshold=-0.5
  And StopperState with best_score=-0.3
  When stopper is called
  Then it returns True (since -0.3 >= -0.5)
```

## Test File
`tests/unit/adapters/stoppers/test_threshold.py`

## Dependencies
- Requires StopperProtocol + StopperState issue to be completed first

## Estimated Scope
~20 lines implementation + ~40 lines tests

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Enables early termination of evolution when target score is achieved, avoiding unnecessary compute once "good enough" performance is reached.

- Add ScoreThresholdStopper class that stops when best_score >= threshold
- Support any float threshold including negative values for loss minimization
- Add comprehensive unit tests covering all acceptance criteria

Test: `uv run pytest tests/unit/adapters/stoppers/test_threshold.py -v`

Closes #193

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [ ] Breaking changes use exclamation mark in title and BREAKING CHANGE: in body

### Review Focus
- ScoreThresholdStopper implementation in `src/gepa_adk/adapters/stoppers/threshold.py`
- Test coverage for edge cases (negative scores, float precision)

### Related
- Parent issue: #51 (Pluggable stop conditions)
- Similar implementation: TimeoutStopper (#192)

### CODE REVIEW

File: src/gepa_adk/adapters/stoppers/threshold.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        Examples:
+            ```python
+            stopper = ScoreThresholdStopper(0.9)  # 90% target
+            stopper = ScoreThresholdStopper(-0.5)  # For negative score domains
+            ```
+
+        Note:
+            Unlike timeout values, threshold has no restrictions on sign or
+            magnitude since score domains vary by application.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/gepa_adk/adapters/stoppers/threshold.py]: Fixed. Updated the TACOS acrostic pattern:
- Class Note now starts with "Any float value..." (A)
- `__init__` Note now starts with "Compared to timeout values..." (C)
